configure.ac

```diff
@@ -1529,55 +1615,45 @@ else
 fi
 #
 hasError=no
-# We need to be careful about PM's that have either conflicting
-# requirements (e.g., different PMI implementations) or different
-# optional features (e.g., MPID_PM_NAMESERVER).
-# In addition, we need to interleave the setup of the PMI and PM
-# modules.  The order is as follows:
-#
-# For each PM, execute the mpichprereq script for that pm (if present).
-# This script provides information about the PM, including which PMI
-# implementations are supported.
-#
-# Then, for the selected PMI, the setup script (if any) is run.  This is
-# necessary because the setup of the PM may require information discovered
-# or provided duing the PMI setup step.
-#
-# Finally, for each PM, the setup script is executed.
-#
-# Step 1: invoke the mpichprereq for each PM
+
 for pm_name in $pm_names ; do
     if test -z "$first_pm_name" ; then
        first_pm_name=$pm_name
        export first_pm_name
     fi
-    if test ! -d $use_top_srcdir/src/pm/$pm_name ; then
-        AC_MSG_WARN([$use_top_srcdir/src/pm/$pm_name does not exist. PM is unknown])
-	hasError=yes
-    elif test ! -x $use_top_srcdir/src/pm/$pm_name/configure -a \
-              ! -f $use_top_srcdir/src/pm/$pm_name/subconfigure.m4 ; then
-        if test -s $use_top_srcdir/src/pm/$pm_name/configure ; then
-	    AC_MSG_WARN([The configure in $use_top_srcdir/src/pm/$pm_name exists but is not executable])
-        else
-            AC_MSG_WARN([pm $pm_name has no configure or subconfigure.m4])
+
+    case "$pm_name" in
+      gforker)
+        build_gforker=yes
+        if test "$with_pmi" != "pmi1" ; then
+            AC_MSG_ERROR([$pm_name requires PMI-v1, but $with_pmi was chosen.])
+        fi
+        ;;
+      remshell)
```

Comment:
nitpick in the commit message i think gforker2 is supposed to be remshell
```
This commit makes 3 orthogonal options:
    * --with-pmi={pmi1, pmi2, pmix}
    * --with-pmilib={mpich, slurm, cray, pmix}
    * --with-pm={no,hydra,hydra2,gforker,gforker2}
```

Reply:
Fixed.